lib/src/client/grpc_or_grpcweb_channel_web.dart

```diff
@@ -23,11 +23,7 @@ class GrpcOrGrpcWebClientChannel extends GrpcWebClientChannel {
     required int grpcWebPort,
```

Comment:
Maybe we should consider changing this constructor to simply be: 

```dart
GrpcOrGrpcWebClientChannel.toSeparateEndpoints({
  required String grpcHost,
  required int grpcPort,
  required String grpcWebHost,
  required int grpcWebPort,
}) ...;

GrpcOrGrpcWebClientChannel({
  required String host,
  required int port
}) : this.toSeparateEndpoints(...);  // Pass the same host and port.
```

Then if server can serve both then you just golden, otherwise you use `toSeparateEndpoints` (and we allow host to be configurable as well). 

WDYT?

Reply:
Your solution is more flexible in general, but I also think that same-host+2-different-ports is the most common setup, so it would probably be nice to provide constructor for it, I think.

Nevertheless, I think this should not be a part of this PR, which basically aims to fix #482
I've created a [separate issue for this](https://github.com/grpc/grpc-dart/issues/484) and will prepare some draft PR (tomorrow probably) so we can discuss it there.